src/shared/components/mutationTable/column/ascnCopyNumber/ASCNCopyNumberColumnFormatter.tsx

```diff
@@ -0,0 +1,195 @@
+import * as React from 'react';
+import { ClonalValue } from 'shared/components/mutationTable/column/clonal/ClonalColumnFormatter';
+import ClonalColumnFormatter from 'shared/components/mutationTable/column/clonal/ClonalColumnFormatter';
+import { ClinicalData } from 'shared/api/generated/CBioPortalAPI';
+import { Mutation } from 'shared/api/generated/CBioPortalAPI';
+import { hasASCNProperty } from 'shared/lib/MutationUtils';
+import SampleManager from 'pages/patientView/SampleManager';
+import ASCNCopyNumberElement from 'shared/components/mutationTable/column/ascnCopyNumber/ASCNCopyNumberElement';
+import { ASCNCopyNumberValue } from 'shared/components/mutationTable/column/ascnCopyNumber/ASCNCopyNumberElement';
+import { ASCN_BLACK } from 'shared/lib/Colors';
+import { getASCNCopyNumberColor } from 'shared/lib/ASCNUtils';
+
+/**
+ * @author Avery Wang
+ */
+
+// gets value displayed in table cell - "NA" if missing attributes needed for calculation
+function getAscnCopyNumberData(
+    mutation: Mutation,
+    sampleIdToClinicalDataMap:
+        | { [sampleId: string]: ClinicalData[] }
+        | undefined
+) {
+    return hasASCNProperty(mutation, 'ascnIntegerCopyNumber')
+        ? mutation.alleleSpecificCopyNumber.ascnIntegerCopyNumber
+        : ASCNCopyNumberValue.NA;
+}
+
+// sort by total copy number (since that is the number displayed in the icon
+function getAllTotalCopyNumberForMutation(
+    data: Mutation[],
+    sampleIdToClinicalDataMap: { [key: string]: ClinicalData[] } | undefined,
+    sampleIds: string[]
+) {
+    const sampleToCNA: { [key: string]: string } = {};
+    for (const mutation of data) {
```

Comment:
this looks like should be another keyBy and then mapValues operation.    it will make the code clearer if the control logic models the transformation

Reply:
@alisman is there any example of `keyBy/mapValues` operations in the code you could point me to? Thanks!

Reply:
added!